### PR TITLE
fix prefer-template violations (4)

### PR DESCRIPTION
**Pull request by lionralfs**
#### Summary
Fixes violations of the `prefer-template` ESLint rule for code blocks in `web/api`
See https://github.com/orgs/mdn/discussions/143#discussioncomment-3193114

This PR…
-->
- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [x] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/api/namednodemap/length/index.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ A number containing the number of objects in the map
 const pre = document.getElementsByTagName("pre")[0];
 const attrMap = pre.attributes;
 
-let result = "The `test` attribute contains " + attrMap.length + " attributes.\n";
+let result = `The \`test\` attribute contains ${attrMap.length} attributes.\n`;
```
Comment (teoli2003) [inline files/en-us/web/api/namednodemap/length/index.md:30]:
```suggestion
let result = `The 'test' attribute contains ${attrMap.length} attributes.\n`;
```

File: files/en-us/web/api/namednodemap/removenameditem/index.md [hunk 2]
```diff
@@ -42,7 +42,7 @@ The removed {{domxref("Attr")}}.
 const pre = document.getElementsByTagName("pre")[0];
 const attrMap = pre.attributes;
 
-let result = "The `test` attribute initially contains `" + attrMap["test"].value + "`.\n";
+let result = `The \`test\` attribute initially contains \`${attrMap["test"].value}\`.\n`;
```
Comment (teoli2003) [inline files/en-us/web/api/namednodemap/removenameditem/index.md:45]:
```suggestion
let result = `The 'test' attribute initially contains '${attrMap["test"].value}'.\n`;
```

File: files/en-us/web/api/namednodemap/removenameditemns/index.md [hunk 3]
```diff
@@ -51,7 +51,7 @@ const pre = document.getElementsByTagName("pre")[0];
 const warning = doc.getElementsByTagName("warning")[0];
 const attrMap = warning.attributes;
 
-let result = "The `ob:one` attribute initially contains `" + attrMap["ob:one"].value + "`.\n";
+let result = `The \`ob:one\` attribute initially contains \`${attrMap["ob:one"].value}\`.\n`;
```
Comment (teoli2003) [inline files/en-us/web/api/namednodemap/removenameditemns/index.md:54]:
```suggestion
let result = `The 'ob:one' attribute initially contains '${attrMap["ob:one"].value}'.\n`;
```

File: files/en-us/web/api/namednodemap/setnameditem/index.md [hunk 4]
```diff
@@ -46,20 +46,20 @@ const span = document.getElementsByTagName("span")[0];
 const pre = document.getElementsByTagName("pre")[0];
 const attrMap = pre.attributes;
 
-let result = "The `<pre>` element initially contains " + attrMap.length + " attributes.\n\n";
+let result = `The \`<pre>\` element initially contains ${attrMap.length} attributes.\n\n`;
```
Comment (teoli2003) [inline files/en-us/web/api/namednodemap/setnameditem/index.md:49]:
```suggestion
let result = `The '<pre>' element initially contains ${attrMap.length} attributes.\n\n`;
```

File: files/en-us/web/api/namednodemap/setnameditem/index.md [hunk 5]
```diff
@@ -46,20 +46,20 @@ const span = document.getElementsByTagName("span")[0];
...
 
 result += "We remove `one` from `<span>` and adds it to `<pre>`.\n";
 const one = span.attributes.removeNamedItem("one");
 attrMap.setNamedItem(one);
-result += "The `<pre>` element now contains " + pre.attributes.length + " attributes.\n\n";
+result += `The \`<pre>\` element now contains ${pre.attributes.length} attributes.\n\n`;
```
Comment (teoli2003) [inline files/en-us/web/api/namednodemap/setnameditem/index.md:54]:
```suggestion
result += `The '<pre>' element now contains ${pre.attributes.length} attributes.\n\n`;
```

File: files/en-us/web/api/namednodemap/setnameditemns/index.md [hunk 6]
```diff
@@ -56,16 +56,16 @@ const pre = document.getElementsByTagName("pre")[0];
 const warning = doc.getElementsByTagName("warning")[0];
 const attrMap = span.attributes;
 
-let result = "The `<span>` element initially contains " + attrMap.length + " attribute.\n\n";
+let result = `The \`<span>\` element initially contains ${attrMap.length} attribute.\n\n`;
```
Comment (teoli2003) [inline files/en-us/web/api/namednodemap/setnameditemns/index.md:59]:
```suggestion
let result = `The '<span>' element initially contains ${attrMap.length} attribute.\n\n`;
```

File: files/en-us/web/api/namednodemap/setnameditemns/index.md [hunk 7]
```diff
@@ -56,16 +56,16 @@ const pre = document.getElementsByTagName("pre")[0];
...
 
 result += "We remove `one` from `<span>` and adds it to `<pre>`.\n";
 const one = warning.attributes.removeNamedItemNS("http://www.example.com/ob", "one");
 attrMap.setNamedItemNS(one);
-result += "The `<span>` element now contains " + span.attributes.length + " attributes:\n\n";
+result += `The \`<span>\` element now contains ${span.attributes.length} attributes:\n\n`;
```
Comment (teoli2003) [inline files/en-us/web/api/namednodemap/setnameditemns/index.md:64]:
```suggestion
result += `The '<span>' element now contains ${span.attributes.length} attributes:\n\n`;
```

File: files/en-us/web/api/namednodemap/setnameditemns/index.md [hunk 8]
```diff
@@ -56,16 +56,16 @@ const pre = document.getElementsByTagName("pre")[0];
 const warning = doc.getElementsByTagName("warning")[0];
 const attrMap = span.attributes;
 
-let result = "The `<span>` element initially contains " + attrMap.length + " attribute.\n\n";
+let result = `The \`<span>\` element initially contains ${attrMap.length} attribute.\n\n`;
 
 result += "We remove `one` from `<span>` and adds it to `<pre>`.\n";
```
Comment (teoli2003) [inline files/en-us/web/api/namednodemap/setnameditemns/index.md:61]:
```suggestion
result += "We remove `one` from '<span>' and adds it to '<pre>'.\n";
```

File: files/en-us/web/api/namednodemap/setnameditem/index.md [hunk 9]
```diff
@@ -46,20 +46,20 @@ const span = document.getElementsByTagName("span")[0];
...
 
 result += "We get `two` from `<span>` and try to adds it to `<pre>`.\n";
```
Comment (teoli2003) [inline files/en-us/web/api/namednodemap/setnameditem/index.md:56]:
```suggestion
result += "We get 'two' from '<span>' and try to adds it to '<pre>'.\n";
```

File: files/en-us/web/api/namednodemap/getnameditem/index.md [hunk 10]
```diff
@@ -40,7 +40,7 @@ An {{domxref("Attr")}} corresponding to the `name` given in parameter, or `null`
 const pre = document.getElementsByTagName("pre")[0];
 const attrMap = pre.attributes;
 
-let result = "The `test` attribute contains " + attrMap.getNamedItem("test").value + "\n";
+let result = `The \`test\` attribute contains ${attrMap.getNamedItem("test").value}\n`;
```
Comment (teoli2003) [inline files/en-us/web/api/namednodemap/getnameditem/index.md:43]:
```suggestion
let result = `The 'test' attribute contains ${attrMap.getNamedItem("test").value}\n`;
```

File: files/en-us/web/api/namednodemap/getnameditem/index.md [hunk 11]
```diff
@@ -40,7 +40,7 @@ An {{domxref("Attr")}} corresponding to the `name` given in parameter, or `null`
...
 if (attrMap["boum"]) {
   result += "And `boum` has been found.";
```
Comment (teoli2003) [inline files/en-us/web/api/namednodemap/getnameditem/index.md:45]:
```suggestion
  result += "And 'boum' has been found.";
```

File: files/en-us/web/api/namednodemap/getnameditemns/index.md [hunk 12]
```diff
@@ -45,8 +45,7 @@ const warning = doc.getElementsByTagName("warning")[0];
 
 const attrMap = warning.attributes;
 
-pre.textContent = "The `ob:one` attribute contains: " +
-                  attrMap.getNamedItemNS("http://www.example.com/ob", "one").value + ".";
+pre.textContent = `The \`ob:one\` attribute contains: ${attrMap.getNamedItemNS("http://www.example.com/ob", "one").value}.`;
```
Comment (teoli2003) [inline files/en-us/web/api/namednodemap/getnameditemns/index.md:48]:
```suggestion
pre.textContent = `The 'ob:one' attribute contains: ${attrMap.getNamedItemNS("http://www.example.com/ob", "one").value}.`;
```

File: files/en-us/web/api/namednodemap/length/index.md (hunk 1 above)
Comment (teoli2003) [inline files/en-us/web/api/namednodemap/length/index.md:30]:
```suggestion
let result = `The 'test' attribute contains ${attrMap.length} attributes.\n`;
```

**Review by teoli2003 — COMMENTED** [review]
Just removing the backtick as a marker inside the string (only place we do this)

File: files/en-us/web/api/navigator/getusermedia/index.md [hunk 13]
```diff
@@ -99,7 +99,7 @@ if (navigator.getUserMedia) {
          };
       },
       function(err) {
-         console.log("The following error occurred: " + err.name);
+         console.log(`The following error occurred: ${err.name}`);
```
Comment (teoli2003) [inline files/en-us/web/api/navigator/getusermedia/index.md:102]:
```suggestion
         console.error(`The following error occurred: ${err.name}`);
```

File: files/en-us/web/api/offlineaudiocontext/index.md [hunk 14]
```diff
@@ -100,7 +100,7 @@ function getData() {
           song.start();
         }
       }).catch(function(err) {
-          console.log('Rendering failed: ' + err);
+          console.log(`Rendering failed: ${err}`);
```
Comment (teoli2003) [inline files/en-us/web/api/offlineaudiocontext/index.md:103]:
```suggestion
          console.error(`Rendering failed: ${err}`);
```

File: files/en-us/web/api/offlineaudiocontext/startrendering/index.md [hunk 15]
```diff
@@ -99,7 +99,7 @@ function getData() {
           song.start();
         }
       }).catch(function(err) {
-          console.log('Rendering failed: ' + err);
+          console.log(`Rendering failed: ${err}`);
```
Comment (teoli2003) [inline files/en-us/web/api/offlineaudiocontext/startrendering/index.md:102]:
```suggestion
          console.error(`Rendering failed: ${err}`);
```

File: files/en-us/web/api/pointer_events/multi-touch_interaction/index.md [hunk 16]
```diff
@@ -241,10 +241,11 @@ function enableLog(ev) {
...
   const o = document.getElementsByTagName('output')[0];
-  const s = name + ": pointerID = " + ev.pointerId +
-                " ; pointerType = " + ev.pointerType +
-                " ; isPrimary = " + ev.isPrimary;
-  o.innerHTML += s + "<br>";
+  const s = `${name}:<br>`;
+  s += `  pointerID   = ${ev.pointerId}<br>`;
+  s += `  pointerType = ${ev.pointerType}<br>`;
+  s += `  isPrimary   = ${ev.isPrimary}`;
+  o.innerHTML += `${s}<br>`;
```
Comment (teoli2003) [inline files/en-us/web/api/pointer_events/multi-touch_interaction/index.md:248]: This one is incorrect. `s` is const.

File: files/en-us/web/api/pointer_events/pinch_zoom_gestures/index.md [hunk 17]
```diff
@@ -207,10 +207,11 @@ function enableLog(ev) {
...
   const o = document.getElementsByTagName('output')[0];
-  const s = prefix + ": pointerID = " + ev.pointerId +
-                " ; pointerType = " + ev.pointerType +
-                " ; isPrimary = " + ev.isPrimary;
-  o.innerHTML += s + "<br>";
+  const s = `${name}:<br>`;
+  s += `  pointerID   = ${ev.pointerId}<br>`;
+  s += `  pointerType = ${ev.pointerType}<br>`;
+  s += `  isPrimary   = ${ev.isPrimary}`;
+  o.innerHTML += `${s}<br>`;
```
Comment (teoli2003) [inline files/en-us/web/api/pointer_events/pinch_zoom_gestures/index.md:214]: Same here

**Review by teoli2003 — CHANGES_REQUESTED** [review]
Two cases where the update leads to a syntax error (or similar)

File: files/en-us/web/api/pointer_events/multi-touch_interaction/index.md (hunk 16 above)
Comment (teoli2003) [reply files/en-us/web/api/pointer_events/multi-touch_interaction/index.md:248]: I'll keep a `+` rather than assignment operators.

Comment (lionralfs) [reply files/en-us/web/api/pointer_events/multi-touch_interaction/index.md:248]: Good catch, I'll commit a fix

**Review by teoli2003 — APPROVED** [review]
Thanks!

**Pull request merged**